apollo-router/src/plugins/telemetry/apollo_exporter.rs

```diff
@@ -163,6 +168,15 @@ impl ApolloExporter {
     }
 
     pub(crate) async fn submit_report(&self, report: Report) -> Result<(), ApolloExportError> {
+        // If studio has previously told us not to submit reports, throw it away
+        if Instant::now() < *STUDIO_BACKOFF.lock().unwrap() {
+            tracing::info!(
+                "studio is not accepting reports, discarding report containing {} operations",
```

Comment:
Suspect that it's not OK to discard operations.

Reply:
I agree, but buffering requires some thinking.

We could have a ring buffer with a configurable capacity. How does that sound?

Reply:
Resolved by modifying approach to preserve metrics reports and report tracing error submission backoffs as errors.